### DISCUSSION TITLE
Will there be official support for cypress component testing with ionic angular apps and monorepo?

### DISCUSSION THREAD

**Discussion by ilsehaydee22** [cypress-io/cypress#26977] [Questions and Help]
I've looked into another issue very similar to mine https://github.com/cypress-io/cypress/issues/24226
Where someone mentions that there isn't official Cypress support for ionic apps, I'm aware of that.
The main reason of why I wanted to create this discussion was to ask if cypress planned on giving official support to ionic apps since in the company where I work we are trying to implement component testing to our ionic angular app, and components don't render correctly by passing the component directly like: cy.mount(ExampleComponent). You have to pass in the html in order to render the component, 
![image](https://github.com/cypress-io/cypress/assets/135156367/45cc1897-a41b-4762-b1d3-b1087b6c7ddb)
is the only workaround I've found so far thanks to the other issue where they mention this. We are trying to implement component testing on the company where I work, but we've been encountering different issues such as that one.
Another question I had for the cypress team was: in our company we have a monorepo with more that 3 angular apps. Applications are split into small libs (shared, features, UI, data access) for now we have plenty of tests with cypress e2e, but we wanted to implement component testing, but I've found very difficult to prepare "cy.mount" with all imports, services and more,  so that it works and mounts the component , since our components are very very big and involves a lot of dependencies. , at the moment I could not make the test work nor mount the component. Do you find convenient using this kind of testing our should we stick to e2e

**Comment by portyu9** [comment]
Hi @ilsehaydee22 

There still does not appear to be a separate, officially supported **Ionic Angular Component Testing adapter** in Cypress. The related issue you opened (#26975) was ultimately closed as `not planned`.

For the monorepo part, though, current Cypress Angular Component Testing has a useful pattern that avoids repeating the entire Angular setup in every spec: create a custom `cy.mount()` command containing the imports/providers/declarations that most of your components need.

For example, conceptually:

```ts
import { mount } from 'cypress/angular'

Cypress.Commands.add('mount', (component, config = {}) => {
  return mount(component, {
    ...config,
    imports: [
      /* common Angular/Ionic imports */
      ...(config.imports || []),
    ],
    providers: [
      /* common providers */
      ...(config.providers || []),
    ],
  })
})
```

Then individual component specs only add dependencies that are specific to that component.

That said, if mounting one component requires most of the application's modules, services, routing, authentication, etc., that is usually a sign that it is not a particularly good **component-test boundary**. In a large monorepo I would use Component Testing primarily for smaller shared/UI/leaf components and keep E2E tests for heavily integrated feature components.

For Ionic specifically, the component also needs the Ionic/Angular bootstrap dependencies and styles that it relies on at runtime. Passing an HTML template may happen to get around the rendering problem, but I would not treat that as a general replacement for mounting the actual component.

So I would not replace your existing E2E suite wholesale. A mixed approach is probably the most maintainable one:

- component tests for isolated shared/UI components,
- a custom `cy.mount()` to centralize common Angular setup,
- E2E tests for components/features that depend heavily on the complete application environment.

Please mark this answer as accepted if it helped, thank you.
